File: app/lib/core/constants/product_type.dart
import 'package:flutter/material.dart';

/// Tipe produk di katalog apotek (selaras dengan API `product_type`).
enum ProductType {
  drug('DRUG', 'Obat'),
  health('HEALTH', 'Produk kesehatan'),
  commercial('COMMERCIAL', 'Produk komersial');

  const ProductType(this.apiValue, this.label);

  final String apiValue;
  final String label;

  static ProductType fromApi(String? value) {
    switch (value?.toUpperCase()) {
      case 'HEALTH':
        return ProductType.health;
      case 'COMMERCIAL':
        return ProductType.commercial;
      default:
        return ProductType.drug;
    }
  }

  bool get canRequirePrescription => this == ProductType.drug;

  IconData get icon {
    switch (this) {
      case ProductType.drug:
        return Icons.medication_outlined;
      case ProductType.health:
        return Icons.health_and_safety_outlined;
      case ProductType.commercial:
        return Icons.shopping_bag_outlined;
    }
  }
}

File: api/src/common/constants/product-type.ts
export enum ProductType {
  DRUG = 'DRUG',
  HEALTH = 'HEALTH',
  COMMERCIAL = 'COMMERCIAL',
}

export const PRODUCT_TYPE_LABELS: Record<ProductType, string> = {
  [ProductType.DRUG]: 'Obat',
  [ProductType.HEALTH]: 'Produk kesehatan',
  [ProductType.COMMERCIAL]: 'Produk komersial',
};

export function normalizeProductType(value?: string): ProductType {
  const upper = value?.trim().toUpperCase();
  if (upper === ProductType.HEALTH) return ProductType.HEALTH;
  if (upper === ProductType.COMMERCIAL) return ProductType.COMMERCIAL;
  return ProductType.DRUG;
}

export function assertPrescriptionAllowed(
  productType: ProductType,
  requiresPrescription?: boolean,
) {
  if (requiresPrescription && productType !== ProductType.DRUG) {
    throw new Error(
      'Butuh resep hanya berlaku untuk tipe produk Obat (DRUG)',
    );
  }
}
